airflow-core/src/airflow/ui/src/utils/sanitizeUrl.ts

```diff
@@ -0,0 +1,84 @@
+/*!
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+type UrlType = "bundle" | "iframe";
+
+/**
+ * Sanitizes URLs for safe use in iframe src attributes or dynamic imports
+ * Prevents XSS by validating protocol and removing dangerous content
+ */
+export const sanitizeUrl = (url: string, type: UrlType): string => {
+  try {
+    // Parse the URL to validate its structure
+    const urlObj = new URL(url);
+
+    // Only allow http, https, and relative URLs
+    if (urlObj.protocol !== "http:" && urlObj.protocol !== "https:" && urlObj.protocol !== "") {
+      throw new Error("Invalid protocol");
```

Comment:
Interesting. One of the hypothesis I had was that, none of the external views or react apps are user provided values. 

Those are made by plugins developer and deployed by deployment manager, so I figured we could trust such values.

WDYT, is that needed in some scenarios ?